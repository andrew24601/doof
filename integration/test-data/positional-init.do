// Test positional object initialization
// Tests: Positional initialization syntax with parentheses

class Point {
    x: int;
    y: int;
}

class Person {
    name: string;
    age: int;
}

function main(): int {
    // Positional initialization for class without constructor
    let point1 = Point(10, 20);
    println(point1.x);      // Should print: 10
    println(point1.y);      // Should print: 20
    
    // Positional initialization for class with constructor
    let person1 = Person("Bob", 25);
    println(person1.name);  // Should print: Bob
    println(person1.age);   // Should print: 25
    
    return 0;
}
