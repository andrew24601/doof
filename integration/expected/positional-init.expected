10
20
Bob
25
